src/town_gui.cpp

```diff
@@ -1584,6 +1606,64 @@ struct BuildHouseWindow : public PickerWindow {
 		}
 	}
 
+	static std::string GetHouseYear(TimerGameCalendar::Year min_year, TimerGameCalendar::Year max_year)
+	{
+		if (min_year == CalendarTime::MIN_YEAR) {
+			return GetString(STR_HOUSE_PICKER_YEARS_ANY);
+		}
```

Comment:
As it currently stands, for a house that is designed to be built only until 1960, the years line will show "Any" instead of the correct phrase of "Until 1960."